### PR TITLE
Add a lookup method with auto padding for nn.Embedding

### PR DESCRIPTION
**Pull request by Gosicfly**
When we use RNN for sequence models, we usually need to pad the item in the batch to a same length. In this PR, I add a method for nn.Embedding, which can make a lookup operation with auto padding. It looks like this:

```
embed = nn.Embedding(11, 4, padding_idx=10)
ids = [[2, 3, 4], [2, 3]]
embeddings, sorted_embeddings, sorted_len = embed.lookup_with_autopadding(ids)

print(embeddings)
# Variable containing:
# (0 ,.,.) = 
#  -1.0276 -0.5631 -0.8923 -0.0583
#  -0.1955 -0.9656  0.4224  0.2673
#  -0.4212 -0.5107 -1.5727 -0.1232
# 
# (1 ,.,.) = 
#  -1.0276 -0.5631 -0.8923 -0.0583
#  -0.1955 -0.9656  0.4224  0.2673
#   0.0000  0.0000  0.0000  0.0000
# [torch.FloatTensor of size 2x3x4]

print(sorted_embeddings)
# Variable containing:
# (0 ,.,.) = 
#  -1.0276 -0.5631 -0.8923 -0.0583
#  -0.1955 -0.9656  0.4224  0.2673
#  -0.4212 -0.5107 -1.5727 -0.1232
# 
# (1 ,.,.) = 
#  -1.0276 -0.5631 -0.8923 -0.0583
#  -0.1955 -0.9656  0.4224  0.2673
#   0.0000  0.0000  0.0000  0.0000
# [torch.FloatTensor of size 2x3x4]

print(sorted_len)
# Variable containing:
#  3
#  2
# [torch.LongTensor of size 2]
```

### CODE REVIEW

**Comment by apaszke** [comment]
I'm not sure if it's really worth including this in the library. Padding lists of indices is a matter of doing
```python
max_len = max(len(seq) for seq in sequences)
sequences = [seq + [pad_idx] * (max_len - len(seq)) for seq in sequences]
```
in your script. I'd err on the side of keeping the core library lean, and maintaining consistency with the fact that modules should be used as `module(input)`, and never as `module.some_method(input)`.

Thanks a lot for the PR, but please consider opening an issue before you prepare the code. This will let us discuss things before you do any work and help avoid situations like this one.

**Comment by soumith** [comment]
i think a better (and practical) proposal might be to have nn.Embedding process `padded_sequence`

**Comment by apaszke** [comment]
Yeah, relaxing the Embedding interface to accept lists of uneven length would be fine as well

**Comment by Gosicfly** [comment]
Got that. tks for reply.

**Comment by Gosicfly** [comment]
Is it worth to add an parameter `autopadding` to nn.Embedding init method?
like:
```
def __init__(self, num_embeddings, embedding_dim, padding_idx=None,
                 max_norm=None, norm_type=2, scale_grad_by_freq=False,
                 sparse=False, autopadding=False):
    pass
```
